### DISCUSSION TITLE
Game stopped working. Game without DFhack works fine. Tryed steam realease and manual install.

### DISCUSSION THREAD

**Discussion by ValkivirIceUse** [DFHack/dfhack#3442] [General]
How could this happen? It was working this morning! Nothing changed :(

**Comment by myk002** [comment]
define "stopped working" -- does it crash on start? or does DFHack pop up a dialog saying the version isn't supported?

**Reply by ValkivirIceUse** [reply]
Dark screen appears for a second and crash after half second. r1 Works altho. So it's something with newest release :/ but why just me?

**Reply by ValkivirIceUse** [reply]
How to change in dfhack files to not speed up game start?

**Reply by ValkivirIceUse** [reply]
Okey, after changing dfhooks.dll game started. Used file from r1 release

**Reply by ValkivirIceUse** [reply]
99%sure that this file make this error. Ofc i tryed admin mode and everything.

**Reply by ValkivirIceUse** [reply]
Unfortunately, experimental and other branches also doesn't work. Reinstalled everything from scratch, nothing works. Which file can give u insight into why my game doesnt work? :( as i have a lot of blueprints and other things i find hard to play without.

100%sure that problem is with dfhooks.dll.

**Reply by ValkivirIceUse** [reply]
Hopefully i have 19may release somewhere over here, so i hope i can rollback and save something... Tell me what should i do, couse maybe i am lacking some software, some files or updates.

**Reply by myk002** [reply]
which branch are you using? the regular DFHack 50.08-r2 or the one for experimental SDL2?

To disable the fast start, edit the `dfhack-config/init/dfhack.control-panel-system.init` file and comment out the line that says `enable faststart`. DF has a race condition bug in the SDL2 version that `faststart` makes more likely to trigger. It shouldn't negatively affect the regular DF 50.08, though.

**Reply by ValkivirIceUse** [reply]
Regular one. Tryed changing to disable faststart. Now i want to add latest release before r2. How can i stop updating the game via steam? Or i try using old method and just copy it over Df folder.

**Reply by ValkivirIceUse** [reply]
Nope, it still crashes.

**Reply by ValkivirIceUse** [reply]
Okey i managed to get back to before r2, still have my commands and everything. Works perfectly. So it's something with plugins and this one file.

**Reply by myk002** [reply]
well, `dfhooks.dll` *is* DFHack, so an error with that file is unfortunately not very specific.

> experimental and other branches also doesn't work

do you mean it still crashes? In particular, if you subscribe to the `experimental` branch of DF and the `beta_experimental_sdl2` of DFHack, *and* disable `faststart`, does it still crash for you?

**Comment by IvoryTower33** [comment]
I'm afraid I'm experiencing the same Issue. Just as it starts to load the client, black screen and crash.

**Reply by lethosor** [reply]
What version numbers of DF and DFHack are you using?

**Comment by ValkivirIceUse** [comment]
DF Stable release steam and tried with other versions and faststart. Faststart was working perfectly in the r2rc1. So the issue comes from something that was added between those two versions. I am afraid that the issue can also comes from our side, as it seems that not so many people HAVE this issue :(. i will play r2rc1 for now, as i've been struggling few hours yesterday and just want to have some fun :D.

**Reply by lethosor** [reply]
Wait, so r2rc1 works for you but r2 does not? Is this the case for anyone else (@IvoryTower33)? There was not that much that changed between the two versions: https://github.com/dfhack/dfhack/compare/50.08-r2rc1...50.08-r2

It's _possible_ that the `Core.cpp`, `DFSteam.cpp`, or `launchdf.cpp` changes could be related. Pretty much everything else is a module/plugin/script that shouldn't be running this early in startup.

This thread has a couple possible ideas for troubleshooting further: https://github.com/DFHack/dfhack/discussions/3442#discussioncomment-6078760

**Reply by IvoryTower33** [reply]
r1 works perfectly fine, yes

**Reply by lethosor** [reply]
What about r2rc1, though?

**Reply by ValkivirIceUse** [reply]
r2rc1 works perfectly as i said ;)

**Reply by lethosor** [reply]
Well, you only mentioned faststart, but my question was actually for @IvoryTower33 - did r2rc1 work for you?

**Reply by IvoryTower33** [reply]
yep

**Comment by ValkivirIceUse** [comment]
Df 50.08

**Comment by ValkivirIceUse** [comment]
Someone replied on steam too, so probably more people will have this issue. the thing is, i do not remember if i run df once on r2 :/. i only remember that i was playing r2rc1, so i updated, and boom. 

What files can contain information about the crash? Is there such file?

**Reply by Stazos** [reply]
Is Dwarf Fortress installed on a different drive than system? I had the same issue and after I reinstalled DFhack and Dwarf fortress on my C drive it started working again.

**Reply by lethosor** [reply]
Can you upload your `stderr.log` file in case it contains anything useful?

**Comment by IvoryTower33** [comment]
DF is installed on main drive (system) Here is the content of stderr.log. I didn't experience this error with r1.

DFHack build: 50.08-r2-0-gecf82471
Starting with working directory: C:\games\Dwarf Fortress 0.58
Identifying DF version.
Loading hack\symbols.xml ... OK
v0.50.08 win64 STEAM (windows): PE: 0x644D7FD2
v0.50.08 win64 ITCH (windows): PE: 0x64504278
v0.50.08 win64 CLASSIC (windows): PE: 0x64504595
Loaded 3 DF symbol tables.
Version: v0.50.08 win64 STEAM
Initializing Console.
Console is running.
Binding to SDL.
Found Steam.

**Comment by myk002** [comment]
Is DF configured to run at full screen or windowed? Does it crash if you switch it to the other setting?

**Reply by IvoryTower33** [reply]
crashes with both settings I'm afraid.

**Comment by myk002** [comment]
After the "Found Steam." log line is output, we attempt to use the Steam API to figure out if we should launch the DFHack Steam monitor process. That is new in -r2 and could very well be the culprit here.

How long have you been using DFHack? 50.08-r1 did *not* use the Steam API, but previous versions (all of the 50.07 era) *did* (back when we had to check whether we were running on Steam Deck and alter console behavior -- now the gui/control-panel preference has obsoleted that code, and it was taken out). 

Are you launching from the Steam client or outside of it? If you are launching from the Steam client, are you launching DFHack or DF? When you launch, could you watch the Steam client carefully and note the order in which the DFHack and DF apps say they are "running", and whether they seem to change from "running" to not running more than once?

Also, what is your operating system version and language?

**Reply by ValkivirIceUse** [reply]
I've been using dfhack before steam release, since alpha, later on i started using steam release. I was always launching dfhack by steam, but i remember i tried opening DF itself too.  System is .. windows 10 Home. Language Polish, but steam client is in English. i will try those methods the other time, as i don't have too much time during the week. And i play r2rc1 for now.

**Comment by IvoryTower33** [comment]
OS: Windows 10 Pro 64 bit 10.0, build 19045. Lang: English
I've been using DFHACK since its first 50v inception and I never experienced anything remotely similar.
I'm launching from outside the Steam client.

Thanks for your awesome work dfhack devs!

**Reply by myk002** [reply]
> I'm launching from outside the Steam client.

This is probably the critical detail. It is not something I can easily test from Linux, since it follows a different code path. Are you running DF from the Steam install directory, or have you copied it somewhere else? Does it work if you delete the steam_api64.dll file?

**Reply by IvoryTower33** [reply]
when I launch it from the steam client it does indeed work

**Reply by ValkivirIceUse** [reply]
You launch DFhack from steam client? and it works? i've been doing that all the time :/

**Reply by myk002** [reply]
@IvoryTower33 @ValkivirIceUse could you possibly install the DFHack build from here: https://github.com/DFHack/dfhack/actions/runs/5181212891#artifacts

and run it so it crashes, then give me the output of the stderr.log file (like in https://github.com/DFHack/dfhack/discussions/3442#discussioncomment-6079646 above)?

**Comment by myk002** [comment]
Proposed fix in #3451

**Comment by lethosor** [comment]
50.08-r3 should fix the crash when launching outside of the Steam client: https://github.com/DFHack/dfhack/releases/tag/50.08-r3

**Comment by ValkivirIceUse** [comment]
It works! Thank you devs :* You're awesome!
